akka-stream/src/main/scala/akka/stream/impl/SubFlowImpl.scala

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.impl
+
+import akka.stream._
+import akka.stream.scaladsl._
+
+import language.higherKinds
+
+object SubFlowImpl {
+  trait MergeBack[In, F[+_]] {
```

Comment:
It’s just a monoid in the category of endofunctors … (no, it is actually a covariant type constructor).